buildman/manager/ephemeral.py

```diff
@@ -330,169 +186,412 @@ def _metric_prefix(self):
-            "had_heartbeat": False,
-            "job_queue_item": build_job.job_item,
+    def start_job(self, job_id, max_build_time):
+        """ Starts the build job. This is invoked by the worker once the job has been created and
+        scheduled, returing the buildpack needed to start the actual build.
+        """
+        try:
+            build_job = self._build_job_from_job_id(job_id)
+        except BuildJobDoesNotExistsError as bjne:
+            logger.warning("Failed to start job %s. Job does not exists - %s", job_id, bjne)
+            return None, None
+        except BuildJobError as bje:
+            logger.error("Failed to get job %s - %s", job_id, bje)
+            return None, None
+
+        # Construct the buildpack
+        repo = build_job.repo_build.repository
+        repository_name = repo.namespace_user.username + "/" + repo.name
+        context, dockerfile_path = build_job.extract_dockerfile_args()
+        base_image_information = {}
+        if build_job.pull_credentials:
+            base_image_information["username"] = build_job.pull_credentials.get("username", "")
+            base_image_information["password"] = build_job.pull_credentials.get("password", "")
+
+        build_args = {
+            "build_package": build_job.get_build_package_url(self._user_files),
+            "context": context,
+            "dockerfile_path": dockerfile_path,
+            "repository": repository_name,
+            "registry": self._registry_hostname,
+            "pull_token": build_job.repo_build.access_token.get_code(),
+            "push_token": build_job.repo_build.access_token.get_code(),
+            "tag_names": build_job.build_config.get("docker_tags", ["latest"]),
+            "base_image": base_image_information,
         }
 
-        lock_payload = json.dumps(payload)
-        logger.debug(
-            "Writing key for job %s with expiration in %s seconds",
-            build_uuid,
-            EPHEMERAL_SETUP_TIMEOUT,
-        )
+        private_key = None
+        if (
+            build_job.repo_build.trigger is not None
+            and build_job.repo_build.trigger.secure_private_key is not None
+        ):
+            private_key = build_job.repo_build.trigger.secure_private_key.decrypt()
+
+        if private_key is not None:
+            build_args["git"] = {
+                "url": build_job.build_config["trigger_metadata"].get("git_url", ""),
+                "sha": build_job.commit_sha(build_job.build_config),
+                "private_key": private_key or "",
+            }
+
+        # If the build args have no buildpack, mark it as a failure before sending
+        # it to a builder instance.
+        if not build_args["build_package"] and not build_args["git"]:
+            logger.error(
+                "Failed to start job %s: insufficient build args - No package url or git",
+                job_id,
+            )
+            # TODO: callback job_completed for cleanup
```

Comment:
Should this be included in the current Pull Request -- or do you expect it to come later?

Reply:
Updated